From 479070d4e13e82b8073f2ac7e1576acbaa5722f1 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 31 Jan 2022 10:23:54 +0200
Subject: [PATCH] [SW-71244] gaudi2: fwl proto spsram to elbi

The FWL protocol include a commands HL_UPD_{MAP,RUN}_SPSRAM
Since gaudi2, the SPSRAM is not used anymore. A more descriptive
name should be HL_UPD_{MAP,RUN}_ELBI

Pay attention, involve changing the specs

Change-Id: Idacf6248b605219625abaaa85a709c4f5d1140e5
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index e5bf7af..4bfddf3 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -330,13 +330,13 @@
 	case HL_UPD_NOOP:
 		break;
 	case HL_UPD_RUN_SRAM:
-	case HL_UPD_RUN_SPSRAM:
+	case HL_UPD_RUN_ELBI:
 	case HL_UPD_GET_PROTO_STATUS:
 		res = FOUND_SLOW_RETURN;
 		break;
 	case HL_UPD_RST_STATE:
 	case HL_UPD_MAP_SRAM:
-	case HL_UPD_MAP_SPSRAM:
+	case HL_UPD_MAP_ELBI:
 	default:
 		res = FOUND_FAST_RETURN;
 		break;
@@ -379,9 +379,9 @@
 	case HL_UPD_CLR_STS:
 	case HL_UPD_RST_STATE:
 	case HL_UPD_MAP_SRAM:
-	case HL_UPD_MAP_SPSRAM:
+	case HL_UPD_MAP_ELBI:
 	case HL_UPD_RUN_SRAM:
-	case HL_UPD_RUN_SPSRAM:
+	case HL_UPD_RUN_ELBI:
 	case HL_UPD_GET_PROTO_STATUS:
 		res = FOUND_FAST_RETURN;
 		break;
@@ -492,13 +492,13 @@
 		module_fwupd_send_status(state, HL_UPD_STS_RST_OK, NULL);
 		break;
 	case HL_UPD_MAP_SRAM:
-	case HL_UPD_MAP_SPSRAM:
+	case HL_UPD_MAP_ELBI:
 		sts = module_fwupd_map(state,
 				       state->wfc_cmd.size, state->wfc_cmd.cmd);
 		module_fwupd_send_status(state, sts, state->map_offset);
 		break;
 	case HL_UPD_RUN_SRAM:
-	case HL_UPD_RUN_SPSRAM:
+	case HL_UPD_RUN_ELBI:
 	case HL_UPD_GET_PROTO_STATUS:
 	default:	/* should never get here */
 		preboot_puts_raw(__func__);
@@ -530,8 +530,8 @@
 	case HL_UPD_MAP_SRAM:
 		cmd_str = "map_sram";
 		/* no break */
-	case HL_UPD_MAP_SPSRAM:
-		cmd_str = cmd_str ?: "map_spsram";
+	case HL_UPD_MAP_ELBI:
+		cmd_str = cmd_str ?: "map_elbi";
 		/* Protect from multi-MAP */
 		if (state->map_offset || state->map_size) {
 			sts = HL_UPD_STS_VALID_ERR;
@@ -546,8 +546,8 @@
 	case HL_UPD_RUN_SRAM:
 		cmd_str = "run_sram";
 		/* no break */
-	case HL_UPD_RUN_SPSRAM:
-		cmd_str = cmd_str ?: "run_spsram";
+	case HL_UPD_RUN_ELBI:
+		cmd_str = cmd_str ?: "run_elbi";
 		/* Protect from RUN before MAP */
 		if (state->map_offset && state->map_size) {
 			sts = HL_UPD_STS_ACK;
@@ -681,7 +681,7 @@
 
 	err = module_fwupd_load_type(state);
 
-	if (cmd == HL_UPD_RUN_SPSRAM)
+	if (cmd == HL_UPD_RUN_ELBI)
 		memset(state->map_offset, 0, ROUND_UP4(state->map_size));
 
 	if (err) {
@@ -720,7 +720,7 @@
 
 	switch (state->wfc_cmd.cmd) {
 	case HL_UPD_RUN_SRAM:
-	case HL_UPD_RUN_SPSRAM:
+	case HL_UPD_RUN_ELBI:
 		err = module_fwupd_run_cmd(state, state->wfc_cmd.cmd, &sts);
 		module_fwupd_send_status(state, sts, NULL);
 		break;
